mcx/distributions/constraints.py

```diff
@@ -175,14 +177,52 @@ def __call__(self, x):
         return np.all(x > 0, axis=-1) & (x_sum <= 1) & (x_sum > 1 - 1e-6)
 
 
+class _IndependentConstraint(Constraint):
+    def __str__(self):
+        return "an event is valid only if all its independent entries are valid."
+
+    def __init__(self, base_constraint, ndims):
+        assert isinstance(base_constraint, Constraint)
+        assert isinstance(ndims, int)
+        assert ndims >= 0
+
+        self.base_constraint = base_constraint
+        self.ndims = ndims
+        super().__init__()
+
+    def __call__(self, value):
+        result = self.base_constraint(value)
+        if self.ndims == 0:
+            return result
+        elif np.ndim(result) < self.ndims:
+            raise ValueError(
+                (f"Expected value.dim() >= {self.ndims}" " but got {np.ndims{result}}")
+            )
+        result = result.all(-1)
+        return result
+
+
+class _PositiveDefinite(Constraint):
```

Comment:
nitpicking: can you add a __str__ method?

Reply:
Ah yes! For sure